src/ports/postgres/modules/utilities/path.py_in

```diff
@@ -155,15 +158,26 @@ def path(schema_madlib, source_table, output_table, partition_expr,
         p_col_orig_name_str = ','.join(
             [i + " AS " + j for i, j in zip(split_p_cols, quoted_split_p_cols)])
         if agg_func:
-            plpy.execute("""
-                CREATE TABLE {output_table} AS
-                   SELECT
-                        {p_col_orig_name_str},
-                        {match_id_name},
-                        {agg_func}
-                   FROM {matched_rows}
-                   GROUP BY {as_partition_expr}, {match_id_name}
-                """.format(**locals()))
+            if partition_expr == '1 = 1':
+                # no partition
+                plpy.execute("""
+                    CREATE TABLE {output_table} AS
+                       SELECT
+                            {match_id_name},
+                            {agg_func}
+                       FROM {matched_rows}
+                       GROUP BY {match_id_name}
+                    """.format(**locals()))
+            else:
+                plpy.execute("""
+                    CREATE TABLE {output_table} AS
+                       SELECT
+                            {p_col_orig_name_str},
+                            {match_id_name},
+                            {agg_func}
+                       FROM {matched_rows}
+                       GROUP BY {partition_expr}, {match_id_name}
+                    """.format(**locals()))
```

Comment:
I presume this is not something specific for overlapping patterns, right?

I guess we can now remove "as_partition_expr" from this function, and
the else clause a little earlier in the function:
if not partition_expr:
            partition_expr = "1 = 1"
            as_partition_expr = "1 = 1 as all"
else:
            as_partition_expr = partition_expr